File: androidApp/src/main/java/com/medhelp/medhelp/data/model/settings/SettingsAllBranchHospitalResponse.java
//package com.medhelp.medhelp.data.model.settings;
//
//import com.google.gson.annotations.SerializedName;
//
//public class SettingsAllBranchHospitalResponse implements Comparable {
//
//    @SerializedName("id_filial") private int idBranch;
//    @SerializedName("naim_filial") private String nameBranch;
//    @SerializedName("shopid") private String idShop;
//    @SerializedName("secretkey") private String keyShope;
//    @SerializedName("clientkey") private String keyAppYandex;
//
//
//    private Boolean isFavorite = false;
//
//    public SettingsAllBranchHospitalResponse() {
//    }
//
//    public SettingsAllBranchHospitalResponse(int idBranch, String nameBranch, Boolean isFavorite) {
//        this.idBranch = idBranch;
//        this.nameBranch = nameBranch;
//        this.isFavorite = isFavorite;
//    }
//
//    public String getIdShop() {
//        return idShop;
//    }
//
//    public void setIdShop(String idShop) {
//        this.idShop = idShop;
//    }
//
//    public String getKeyShope() {
//        return keyShope;
//    }
//
//    public void setKeyShope(String keyShope) {
//        this.keyShope = keyShope;
//    }
//
//    public String getKeyAppYandex() {
//        return keyAppYandex;
//    }
//
//    public void setKeyAppYandex(String keyAppYandex) {
//        this.keyAppYandex = keyAppYandex;
//    }
//
//    public Boolean getFavorite() {
//        return isFavorite;
//    }
//
//    public void setFavorite(Boolean favorite) {
//        isFavorite = favorite;
//    }
//
//    public int getIdBranch() {
//        return idBranch;
//    }
//
//    public void setIdBranch(int idBranch) {
//        this.idBranch = idBranch;
//    }
//
//    public String getNameBranch() {
//        return nameBranch;
//    }
//
//    public void setNameBranch(String nameBranch) {
//        this.nameBranch = nameBranch;
//    }
//
//    @Override
//    public int compareTo(Object o) {
//
//        int res1 = isFavorite.compareTo(((SettingsAllBranchHospitalResponse) o).getFavorite());
//        res1 *= -1;
//
//        if (res1 != 0)
//            return res1;
//        else
//            return nameBranch.compareTo(((SettingsAllBranchHospitalResponse) o).getNameBranch());
//    }
//}

File: shared/src/commonMain/kotlin/com/medhelp/newmedhelp/model/SettingsAllBranchHospitalResponse.kt
package com.medhelp.shared.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class SettingsAllBranchHospitalResponse  : Comparable<SettingsAllBranchHospitalResponse>{
    @SerialName("id_filial")
    var idBranch: Int? = null

    @SerialName("naim_filial")
    var nameBranch: String? = null

    @SerialName("shopid")
    var idShop: String? = null

    @SerialName("secretkey")
    var keyShope: String? = null

    @SerialName("clientkey")
    var keyAppYandex: String? = null

    var isFavorite = false

    override fun compareTo(other: SettingsAllBranchHospitalResponse): Int {
        var res1: Int = isFavorite.compareTo(other.isFavorite)
        res1 *= -1

        return if (res1 != 0) res1 else nameBranch!!.compareTo(other.nameBranch!!)
    }
}